pydocstringformatter/run.py

```diff
@@ -85,7 +88,15 @@ def _format_file(self, filename: Path) -> bool:
                 filename_str = str(filename)
 
             if self.config.write:
-                with open(filename, "w", encoding="utf-8") as file:
+                if isinstance(newlines, tuple):
+                    newlines = newlines[0]
```

Comment:
@DanielNoord Your changes and comments look great. My only concern is if we process a utf-16, or non-text file, the file would most likely get corrupted. While I like the idea of automatically fixing text files that have non-standard newlines, we don't have any logic here to see if the file is really a text file, or not. Would checking for characters `\x00-\x08,\x0b-\x0c,\x0e-\x1f` tell us if the file is a binary file or not?

Perhaps I'm being overly cautious, but I think it's best to avoid corrupting files as much as possible, as the user may not even know the file is non-standard, or accidentally fed in the wrong files. Your thoughts?

Reply:
Although that's a valid point I wonder how often such issues will occur in the wild. Either way, I would recommend to use non-write mode (so leave the `-w`) for any first time user or anybody who is not using a source control program.
If you're using `git` it's not too bad that one file is corrupted as you'll be able to easily revert anyway.

I think the benefit of being careful for very particular and unlikely examples in this case doesn't really weigh up against the benefits of keeping this a bit a simpler and doing some auto-formatting.

_Perhaps we should add the stuff about using `-w` with source control in the README._

Reply:
Thanks again for your help, and thoughtful changes. If it's ever decided to make it "binary aware", I confirmed that dos2unix [uses](https://sourceforge.net/p/dos2unix/dos2unix/ci/master/tree/dos2unix/common.c#l2034) the characters `\x00-\x08,\x0b,\x0e-\x1f` to determine if a file is binary.